core/src/main/java/org/apache/carbondata/core/scan/result/RowBatch.java

```diff
@@ -100,4 +100,19 @@ public int getSize() {
     counter++;
     return row;
   }
+
+  /**
+   * read next batch
+   *
+   * @return rows
+   */
+  public List<Object[]> nextBatch() {
+    if (!hasNext()) {
+      throw new NoSuchElementException();
+    }
+    List<Object[]> row;
+    row = rows.subList(counter, rows.size());
```

Comment:
where is the batch reading happening now ?

you just have to return rows I think.

Reply:
ok, I return rows directly